span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanPreProcessor.java

```diff
@@ -91,14 +94,27 @@ PreProcessedSpan preProcessSpan(Span span) {
         span.getProcess().getTagsList().stream()
             .collect(Collectors.toMap(t -> t.getKey().toLowerCase(), t -> t, (v1, v2) -> v2));
 
-    if (spanDropManager.shouldDropSpan(span, spanTags, processTags)) {
+    Optional<String> tenantIdMaybe =
+        tenantIdHandler.getAllowedTenantId(span, spanTags, processTags);
+    if (tenantIdMaybe.isEmpty()) {
+      return null;
+    }
+    String tenantId = tenantIdMaybe.get();
+    Event event =
+        buildEvent(tenantId, span, tenantIdHandler.getTenantIdProvider().getTenantIdTagKey());
+
+    if (spanDropManager.shouldDropSpan(span, tenantId)
+        || spanDropManager.shouldDropEvent(event, tenantId)) {
       return null;
     }
 
     // filter tags
-    String tenantId = tenantIdHandler.getAllowedTenantId(span, spanTags, processTags).get();
     Span processedSpan = tagsFilter.apply(tenantId, span);
-    return new PreProcessedSpan(tenantId, processedSpan);
+    return new PreProcessedSpan(
+        tenantId,
+        processedSpan,
+        buildEvent(
```

Comment:
This doesn't seem right. We are calling `buildEvent` twice, once with `span`, and other with `processedSpan`. We can first apply the `tagsFilter` on `span` and then build event for the `processedSpan`. 

And, then pass the event to `shouldDropEvent`